src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -1123,53 +1123,6 @@ public ImmutableArray<SymbolDisplayPart> ToDisplayParts(SymbolDisplayFormat form
 
         #region ISymbol Members
 
-        SymbolKind ISymbol.Kind
```

Comment:
>SymbolKind ISymbol.Kind [](start = 8, length = 23)

It looks like the code was reachable through ISymbol.Kind API and it was producing incorrect result at least for Discard and for ErrorType. It feels like we need to add some tests.

Reply:
Added the tests as discussed.